stacks-signer/src/signer.rs

```diff
@@ -1235,35 +1269,56 @@ impl Signer {
         }
     }
 
-    /// Update the DKG for the provided signer info, triggering it if required
-    pub fn update_dkg(
-        &mut self,
-        stacks_client: &StacksClient,
-        current_reward_cycle: u64,
-    ) -> Result<(), ClientError> {
-        let reward_cycle = self.reward_cycle;
+    /// Refresh DKG value and queue DKG command if necessary
+    pub fn refresh_dkg(&mut self, stacks_client: &StacksClient) -> Result<(), ClientError> {
+        // First check if we should queue DKG based on contract vote state and stackerdb transactions
+        let should_queue = self.should_queue_dkg(stacks_client)?;
+        // Before queueing the command, check one last time if DKG has been
+        // approved. It could have happened after the last call to
+        // `get_approved_aggregate_key` but before the theshold check in
+        // `should_queue_dkg`.
         let old_dkg = self.approved_aggregate_public_key;
         self.approved_aggregate_public_key =
-            stacks_client.get_approved_aggregate_key(reward_cycle)?;
+            stacks_client.get_approved_aggregate_key(self.reward_cycle)?;
         if self.approved_aggregate_public_key.is_some() {
             // TODO: this will never work as is. We need to have stored our party shares on the side etc for this particular aggregate key.
             // Need to update state to store the necessary info, check against it to see if we have participated in the winning round and
             // then overwrite our value accordingly. Otherwise, we will be locked out of the round and should not participate.
             self.coordinator
                 .set_aggregate_public_key(self.approved_aggregate_public_key);
             if old_dkg != self.approved_aggregate_public_key {
-                debug!(
+                warn!(
                     "{self}: updated DKG value to {:?}.",
                     self.approved_aggregate_public_key
                 );
             }
-            return Ok(());
-        };
+            if matches!(self.state, State::OperationInProgress(Operation::Dkg)) {
```

Comment:
Why use `if matches!` instead of `if let`?  I thought clippy hated single branch matching, though maybe he can't tell if it's a macro?